multiqc/modules/stacks/stacks.py

```diff
@@ -0,0 +1,287 @@
+                        paired-ends and call variants.
+                        These numbers are obtained from the `gstacks.log.distribs` file''',
+            plot = table.plot(self.cov_data, self.gsheaders, config_table)
+        )
+        # Write population sumstats table
+        config_table = {
+            'id': 'sumstats_table',
+            'namespace': 'stacks'
+        }
+        self.add_section (
+            name = 'Population summary statistics',
+            anchor = 'stacks-sumstats',
+            description = 'Population statistics as calculated from variant sites found in this run',
+            helptext = '''**Note!** The sample names have the following scheme `<run folder name> | <population ID>`,
+                        where the population ID is defined in the input population map file.
+                        This information is obtained from the Stacks program `population` and the file populations.sumstats_summary.tsv
+                        ''',
+            plot = table.plot(self.sumstats_data, self.sheaders, config_table)
+        )
+        config_distribs = {
+            'id': 'distribs_plot',
+            'namespace': 'stacks',
+            'tt_label': '{point.y} loci, {point.x} samples/SNPs',
+            'data_labels': [
+                {'name': 'Samples per loci', 'ylab': '# loci', 'xlab': '# samples'},
+                {'name': 'SNPs per loci', 'ylab': '# loci', 'xlab': '# SNPs'}
+            ]
+        }
+        self.add_section (
+            name = 'Population plots',
+            anchor = 'stacks-distribs',
+            description = 'Plots showing, 1) the number of loci shared by number of samples and 2) the number of SNPs per sample',
+            helptext = '''The distributions are obtained from the Stacks program `populations` and it's output file `populations.log.distribs`.
+            These numbers are Stacks' post-filtering.''',
+            plot = linegraph.plot([self.distribs_loci, self.distribs_snps], config_distribs)
+
+        )
+
+    def parse_gstacks(self, file_contents, s_name=None, f=None):
+
+        headers = None
+        content = None
+        out_dict = OrderedDict()
+
+        for l in file_contents.splitlines():
+            if l.startswith("sample	n_loci	n_used_fw_reads"):
+                headers = ["n_loci", "n_used_fw_reads", "mean_cov","mean_cov_ns"]
+            elif l.startswith("END effective_coverages_per_sample"):
+                break
+            elif headers is not None:
+                cdict = OrderedDict().fromkeys(headers)
+                content = l.split("\t")
+                for i in range(1, len(content)):
+                    cdict[list(cdict.keys())[i-1]] = content[i]
+                if s_name is not None:
+                    out_dict[s_name+" | "+content[0]] = cdict
+                else:
+                    out_dict[content[0]] = cdict
+
+        return out_dict
+
+    def parse_sumstats(self, file_contents, s_name=None, f=None):
```

Comment:
Better not to let `s_name` default to `None` here as there's no method for replacing that. It'd be a confusing error if this does end up being `None`, so better to let it crash I think.